### PR TITLE
feat: allow trade preview when wallet not connected

### PR DESCRIPTION
**Pull request by Matthews3301**
# Description

https://github.com/balancer-labs/frontend-v2/issues/1103

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Dependency changes
- [ ] Code refactor / cleanup
- [ ] Documentation or wording changes
- [ ] Other

## How should this be tested?

On trade page, attempt different types of trades. There will be a new element showing estimated trade fees. This should be the same as "Preview" modal. The advantage is that the fee is displayed even when a wallet is not connected.

## Visual context

A new line with estimated trade fee. This could potentially use smaller font / grey colour in order to decrease importance.

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have requested at least 2 reviews (If the PR is significant enough, use best judgement here)
- [n/a] I have commented my code where relevant, particularly in hard-to-understand areas
- [n/a] If package-lock.json has changes, it was intentional.
- [x] The base of this PR is `master` if hotfix, `develop` if not

### CODE REVIEW

**Comment by pkattera** [comment]
@Matthews3301 I like being able to show people without connected wallets more information. My preference would be to enable previews for all users, even without connected wallets. For example, we could enable previews for all users, and then ask people to connect wallet as the first step on the preview. 

Screen Shot 2022-08-23 at 1 39 18 PM
screen? 

We've discussed doing this before, but from memory, the code from the trade screen which enables a user to refresh prices when it updates would also need to be transferred across.

**Comment by timjrobinson** [comment]
The preview button isn't working for me in Chrome without a wallet connected. It shows as enabled but doesn't do anything when clicked.

**Comment by timjrobinson** [comment]
I managed to reproduce the error. When the trade has not been calculated the preview button is enabled but you can't click it. The console has `'invalid decimal value` in it. Though I thought last time the preview button didn't work the trade had been calculated. 

2022-08-25-122757_1610x790_scrot


Full stack:
```
"Error: invalid decimal value (argument="value", value="", code=INVALID_ARGUMENT, version=bignumber/5.4.2)
    at h.makeError (https://beta-app-git-feat-trade-fees-on-trade-card-balancer.vercel.app/js/chunk-vendors.403bcd6a.js:1056:54197)
    at h.throwError (https://beta-app-git-feat-trade-fees-on-trade-card-balancer.vercel.app/js/chunk-vendors.403bcd6a.js:1056:54317)
    at h.throwArgumentError (https://beta-app-git-feat-trade-fees-on-trade-card-balancer.vercel.app/js/chunk-vendors.403bcd6a.js:1056:54372)
    at y (https://beta-app-git-feat-trade-fees-on-trade-card-balancer.vercel.app/js/chunk-vendors.403bcd6a.js:1:73269)
    at x.fn (https://beta-app-git-feat-trade-fees-on-trade-card-balancer.vercel.app/js/TradePage.7cfceb39.js:17:109191)
    at x.run (https://beta-app-git-feat-trade-fees-on-trade-card-balancer.vercel.app/js/chunk-vendors.403bcd6a.js:963:15505)
    at get value [as value] (https://beta-app-git-feat-trade-fees-on-trade-card-balancer.vercel.app/js/chunk-vendors.403bcd6a.js:963:24196)
    at Object.$ [as getQuote] (https://beta-app-git-feat-trade-fees-on-trade-card-balancer.vercel.app/js/TradePage.7cfceb39.js:17:107611)
    at Object.G [as getQuote] (https://beta-app-git-feat-trade-fees-on-trade-card-balancer.vercel.app/js/TradePage.7cfceb39.js:17:111214)
    at setup (https://beta-app-git-feat-trade-fees-on-trade-card-balancer.vercel.app/js/TradePage.7cfceb39.js:17:65892)
    at eo (https://beta-app-git-feat-trade-fees-on-trade-card-balancer.vercel.app/js/chunk-vendors.403bcd6a.js:963:73751)
    at Ni (https://beta-app-git-feat-trade-fees-on-trade-card-balancer.vercel.app/js/chunk-vendors.403bcd6a.js:963:70884)
    at Ci (https://beta-app-git-feat-trade-fees-on-trade-card-balancer.vercel.app/js/chunk-vendors.403bcd6a.js:963:70670)
    at R (https://beta-app-git-feat-trade-fees-on-trade-card-balancer.vercel.app/js/chunk-vendors.403bcd6a.js:963:54433)
    at N (https://beta-app-git-feat-trade-fees-on-trade-card-balancer.vercel.app/js/chunk-vendors.403bcd6a.js:963:54327)
    at g (https://beta-app-git-feat-trade-fees-on-trade-card-balancer.vercel.app/js/chunk-vendors.403bcd6a.js:963:51205)
    at k (https://beta-app-git-feat-trade-fees-on-trade-card-balancer.vercel.app/js/chunk-vendors.403bcd6a.js:963:53581)
    at Object.process (https://beta-app-git-feat-trade-fees-on-trade-card-balancer.vercel.app/js/chunk-vendors.403bcd6a.js:963:61518)
    at g (https://beta-app-git-feat-trade-fees-on-trade-card-balancer.vercel.app/js/chunk-vendors.403bcd6a.js:963:51243)
    at k (https://beta-app-git-feat-trade-fees-on-trade-card-balancer.vercel.app/js/chunk-vendors.403bcd6a.js:963:53581)
    at C (https://beta-app-git-feat-trade-fees-on-trade-card-balancer.vercel.app/js/chunk-vendors.403bcd6a.js:963:54136)
    at g (https://beta-app-git-feat-trade-fees-on-trade-card-balancer.vercel.app/js/chunk-vendors.403bcd6a.js:963:51141)
    at x.u [as fn] (https://beta-app-git-feat-trade-fees-on-trade-card-balancer.vercel.app/js/chunk-vendors.403bcd6a.js:963:55022)
    at x.run (https://beta-app-git-feat-trade-fees-on-trade-card-balancer.vercel.app/js/chunk-vendors.403bcd6a.js:963:15505)
    at eo (https://beta-app-git-feat-trade-fees-on-trade-card-balancer.vercel.app/js/chunk-vendors.403bcd6a.js:963:73759)
    at ko (https://beta-app-git-feat-trade-fees-on-trade-card-balancer.vercel.app/js/chunk-vendors.403bcd6a.js:963:75487)"
```

**Comment by Matthews3301** [comment]
@timjrobinson thanks for spotting this! This was due to not enough liquidity for this pair - should be fixed now, with a warning panel.

**Comment by timjrobinson** [comment]
That seems to be fixed now. I've found another bug. Can reproduce by doing trade from ETH -> BAL, waiting for it to load, then click the flip button. When entering BAL directly it doesn't show. This is without a wallet connected. 

2022-08-26-162255_505x637_scrot

**Comment by Matthews3301** [comment]
@timjrobinson thanks, should be fixed now.

**Comment by pkattera** [comment]
Two cases which don't work as intended:

- Connected wallet where the trade input amount exceeds the wallet’s balance.
- User starts with a disconnected wallet and proceeds to the preview screen. Upon connecting their wallet, their trade input amount exceeded their newly connected wallet balancer. 

See here: https://www.loom.com/share/bd4f068c45fe4c6e9dacb188274a1db4

In both these cases, the CTA’s on the preview screen are enabled which results in errors. 

In both these cases, on the preview screen, ideally:
- The ‘Approve token’ or ‘Confirm transaction’ buttons should be disabled.
- There needs to be a warning explaining why the user can’t proceed.

In addition, I think we should move all the warnings that we display on the Trade screen, to the Preview screen. The main reason, is that people who saw a warning, may forget about it while context switching to connect a wallet.

Designs here:

https://www.figma.com/file/RaJwiVlHqra4dQ0cQS3OOu/UI-Tweaks?node-id=1124%3A3566

**Comment by Matthews3301** [comment]
@pkattera The errors and warnings are now present on both the Trade screen and the Preview modal.

**Comment by Matthews3301** [comment]
@pkattera @garethfuller This is now ready for re-review and approval.

File: src/composables/trade/useValidation.ts [hunk 1]
```diff
@@ -52,11 +52,12 @@ export default function useValidation(
   );
 
   const validationStatus = computed(() => {
-    if (!isWalletReady.value) return TradeValidation.NO_ACCOUNT;
+    // if (!isWalletReady.value) return TradeValidation.NO_ACCOUNT;
```
Comment (garethfuller) [inline src/composables/trade/useValidation.ts:55]: Just delete this.

**Review by garethfuller — APPROVED** [review]

**Pull request merged**